tool/templates/cpp2/base.txt

```diff
@@ -0,0 +1,35 @@
+#ifndef {{ header_guard }}
+#define {{ header_guard }}
+
+{%~ match decl_include %}
+	{% when Some with (include) ~%}
```

Comment:
not quite happy with needing a full match here but i guess it's fine. Does seem to be the idiomatic way to do things (https://github.com/djc/askama/issues/257)